sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -552,18 +553,26 @@ public static Read read() {
         .setBigQueryServices(new BigQueryServicesImpl())
         .setParseFn(parseFn)
         .setMethod(Method.DEFAULT)
+        .setFormat(DataFormat.AVRO)
         .build();
   }
 
   @VisibleForTesting
-  static class TableRowParser implements SerializableFunction<SchemaAndRecord, TableRow> {
+  public static class TableRowParser implements SerializableFunction<SchemaAndRecord, TableRow> {
 
     public static final TableRowParser INSTANCE = new TableRowParser();
 
     @Override
     public TableRow apply(SchemaAndRecord schemaAndRecord) {
-      return BigQueryAvroUtils.convertGenericRecordToTableRow(
-          schemaAndRecord.getRecord(), schemaAndRecord.getTableSchema());
+      if (schemaAndRecord.getRecord() != null) {
+        return BigQueryAvroUtils.convertGenericRecordToTableRow(
+            schemaAndRecord.getRecord(), schemaAndRecord.getTableSchema());
+      } else if (schemaAndRecord.getRow() != null) {
+        return BigQueryUtils.toTableRow().apply(schemaAndRecord.getRow());
```

Comment:
This will copy all of the data out of each Beam Row into a `TableRow` instance. Ideally we would return the actual `Row` instances instead so that the underlying data can still be lazily accessed.

I think there are two options here:
1) Address this when we add support for Arrow to `BigQueryTable` for use in SQL. It looks like we currently pass a custom conversion function to `BigQueryIO.read` [here](https://github.com/apache/beam/blob/04498d56f9bc17528d807c6ef02434eed29decf1/sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/bigquery/BigQueryTable.java#L212). We could just make that function return the underlying row when we're reading Arrow.
2) Add a function like `BigQueryIO.readRows` that encapsulates the row-conversion logic for all the different read modes. Then we can just call that function in `BigQueryTable#buildIOReader`

Either option could be addressed later, but I'd like to have a plan and a follow-up jira filed.

Reply:
Added todos linking to [jira](https://issues.apache.org/jira/browse/BEAM-9114) issue, so we don't forget to address this.